Core/Src/can_handler.c

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file can_handler.c
+ * @author Hamza Iqbal
+ * @brief Source file for CAN handler.  
+ * @version 0.1
+ * @date 2023-09-22
+ * 
+ * @copyright Copyright (c) 2023
+ * 
+ */
+
+#include <stdlib.h>
+#include "can.h"
+#include "can_config.h"
+
+/* Initializes the CAN handler */
+void can_handler_init()
+{
+    // Assign memory to the queues
+    can1_incoming = malloc(sizeof(msg_queue));
+    can2_incoming = malloc(sizeof(msg_queue));
+    can1_outgoing = malloc(sizeof(msg_queue));
+    can2_outgoing = malloc(sizeof(msg_queue));
+
+    initializeHashMap(&callback_map);
+
+    for(int i = 0; i < NUM_CALLBACKS; i++)
+    {
+        insertFunction(&calllback_map, can_callbacks[i].messageID, can_callbacks[i].function);
+    }
+
+}
+
+/* Function to add a node to the message queue it is automatically called in the interrupt triggered callback */
+static void enqueue(struct msg_queue* queue, can_msg_t msg)
```

Comment:
Again, don't define this in the header file if external files aren't gonna use it